sources/core/Stride.Core.Mathematics/Matrix.cs

```diff
@@ -622,43 +622,91 @@ public void DecomposeLQ(out Matrix L, out Matrix Q)
         }
 
         /// <summary>
-        /// Decomposes a rotation matrix with the specified yaw, pitch, roll
+        /// Decomposes a rotation matrix with the specified yaw, pitch, roll value (angles in radians).
         /// </summary>
-        /// <param name="yaw">The yaw.</param>
-        /// <param name="pitch">The pitch.</param>
-        /// <param name="roll">The roll.</param>
+        /// <param name="yaw">The yaw component in radians.</param>
+        /// <param name="pitch">The pitch component in radians.</param>
+        /// <param name="roll">The roll component in radians.</param>
+        /// <remarks>
+        /// This rotation matrix can be represented by <b>intrinsic</b> rotations in the order <paramref name="yaw"/>, <paramref name="pitch"/>, then <paramref name="roll"/>.
+        /// <br/>
+        /// Therefore the <b>extrinsic</b> rotations to achieve this matrix is the reversed order of operations,
+        /// ie. Matrix.RotationZ(roll) * Matrix.RotationX(pitch) * Matrix.RotationY(yaw)
+        /// </remarks>
         public void Decompose(out float yaw, out float pitch, out float roll)
         {
-            pitch = MathF.Asin(-M32);
-            if (MathF.Cos(pitch) > MathUtil.ZeroTolerance)
+            // Adapted from 'Euler Angle Formulas' by David Eberly - https://www.geometrictools.com/Documentation/EulerAngles.pdf
```

Comment:
I believe this is a good way of attributing license for this kind of work. @xen2 does this qualify to be added to `THIRD PARTY.md`?

Reply:
Could also be moved to the top of the file next to the copyright notice, let's see what Xen2 has to say in this regard.